tests/test_install_server.sh

```
@@ -119,4 +119,77 @@ grep -q "verified_ok=" "$SCRIPT" && grep -q "verified_warn=" "$SCRIPT"
+# sleep run past _PORT_WAIT. Assert the recompute sits between the curl
+# and the sleep.
+awk '/while.*_port_tries.*do/{p=1; next}
+     p && /curl.*api.health/{c=1; next}
+     p && c && index($0, "_remaining=$(( _port_deadline - SECONDS ))"){found=1}
+     p && c && /sleep 1/{exit}
+     END{exit !found}' "$SCRIPT"
+
+echo "test: port-open loop floors curl --max-time at 1 s"
+# Ensures a future edit cannot weaken the deadline guard and silently
+# disable curl's per-attempt timeout (finding #3 / #4 invariant).
+grep -q '_curl_timeout=$(( _remaining > 1 ? _remaining : 1 ))' "$SCRIPT"
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)